### ISSUE TITLE
Cannot run mnist example with gpu

### ISSUE THREAD

**Issue by pigbug419** [Microsoft/nni#769]
Hello, nni team!
I executed mnist tutorial with GPU, but none of the trials launched.
(`nnictl create --config config_gpu.yml`)
When I submit an experiment with non-GPU code(`config.yml`), it worked.
Is there any way to solve?

Thank you.

**nni Environment**:
- nni version: 0.5.1
- nni mode(local|pai|remote): local
- OS: ubuntu 16.04
- python version: 3.5
- is conda or virtualenv used?: virtualenv
- is running in docker?:no
- Tensorflow version: 1.8(with GPU)

**nnimanager.log**
```
[2019-02-21T03:47:48.908Z] INFO [ 'Datastore initialization done' ]
[2019-02-21T03:47:48.933Z] INFO [ 'Rest server listening on: http://0.0.0.0:8080' ]
[2019-02-21T03:47:48.918Z] INFO [ 'RestServer start' ]
[2019-02-21T03:47:48.922Z] INFO [ 'Construct local machine training service.' ]
[2019-02-21T03:47:48.926Z] INFO [ 'RestServer base port is 8080' ]
[2019-02-21T03:47:51.191Z] INFO [ 'NNIManager setClusterMetadata, key: trial_config, value: {"gpuNum":1,"command":"python3 mnist.py","codeDir":"/tmp/sungwoo/nni/annotation/tmp4h2ywsv_"}' ]
[2019-02-21T03:47:51.207Z] INFO [ 'required GPU number is 1' ]
[2019-02-21T03:47:51.221Z] INFO [ 'Starting experiment: qx7XLBNw' ]
[2019-02-21T03:47:51.232Z] INFO [ 'Change NNIManager status from: INITIALIZED to: RUNNING' ]
[2019-02-21T03:47:51.235Z] INFO [ 'Add event listeners' ]
[2019-02-21T03:47:51.245Z] INFO [ 'Run local machine training service.' ]
[2019-02-21T03:47:51.869Z] ERROR [ 'Read GPU summary failed with error: ',
  { RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
      at Socket.onChildStdout (child_process.js:343:14)
      at Socket.emit (events.js:182:13)
      at addChunk (_stream_readable.js:283:12)
      at readableAddChunk (_stream_readable.js:260:13)
      at Socket.Readable.push (_stream_readable.js:219:10)
      at Pipe.onStreamRead (internal/stream_base_commons.js:94:17) cmd: 'nvidia-smi -q -x' } ]
```

**Comment by leelaylay** [comment]
@pigbug419 
You are using python virtualenv as your local python interpreter, so it is necessary to clain your python path in your config file. 
```yaml
trial:
  codeDir: .
  command: /path/to/virtualenv/python3 example.py
  gpuNum: 1
```

**Comment by pigbug419** [comment]
@sangloupmonoutlook When I use remote mode, I got such an error.
`ERROR: Failed! Error is: {"error":"TrainingService setClusterMetadata timeout. Please check your config file."}`

@leelaylay After I specified my virtualenv's python path, I got a same error 😢

**Comment by leelaylay** [comment]
@pigbug419 
I guess that the local mode works successfully but remote mode does not. 
The timeout error shows that there is something wrong with the network connection between your local machine and your remote cluster. 
My suggestion: check the network connection between your local machine and your remote cluster. Also, check TCP 8080 port and 8081 port are open.

**Comment by scarlett2018** [comment]
@SparkSnail had fixed issue in last release, @QuanluZhang to connect the PR here.

**Comment by QuanluZhang** [comment]
the bug "stdout maxBuffer length exceeded" is fixed by https://github.com/Microsoft/nni/pull/943

**Comment by Guptajakala** [comment]
> @sangloupmonoutlook When I use remote mode, I got such an error.
> `ERROR: Failed! Error is: {"error":"TrainingService setClusterMetadata timeout. Please check your config file."}`
> 
> @leelaylay After I specified my virtualenv's python path, I got a same error 😢

Hi I met the same problem. Did you solve it?

**Comment by scarlett2018** [comment]
@Guptajakala  - which version are you using? Had you tried upgrade NNI to 0.6, there is a fix in 0.6.

**Comment by Guptajakala** [comment]
> @Guptajakala - which version are you using? Had you tried upgrade NNI to 0.6, there is a fix in 0.6.

Thanks for your reply, could you tell me how to check the version? nni.__version does not work.

**Comment by Guptajakala** [comment]
OK, from conda, I found the version is 1.0
